### PR TITLE
Usage processing Product Specification 

### LINKED ISSUE #69: Usage Stats Processing as a Service (RH)
We aim to provide a **Tracking Code (**similar to GoogleAnalytics**),** that can be embedded in repository landing pages, that forwards for views and downloads counts to our **Usage Stats Processing Service**.  

<p></p>

Fill our survey if you think this service can help solve your problems.

<p></p>

<p></p>

**SP5: Adapting our processes and workflows to meet our member's needs**

<p></p>

<p></p>

**WireFrames:** [**https://www.figma.com/file/Fd54mIb3WVYqQfw3KQmp68/SAAS-Processor?node-id=0%3A1**](https://www.figma.com/file/Fd54mIb3WVYqQfw3KQmp68/SAAS-Processor?node-id=0%3A1)

<p></p>

First review: <https://docs.google.com/document/d/1sJUQ_IemyPp0CktKXv_tfa7PhHcZ1NCBNUZwAv0iwCc/edit#bookmark=id.wrlbox7qb41h>

<p></p>

<p></p>

Product Specification: <https://docs.google.com/document/d/1K3JuDq1gqOXef2bDMTvv7YgXBnE1qdp54Rfd0yWmotk/edit#heading=h.37sflf1e0bm9>

### PR DESCRIPTION
**Pull request by kjgarza**
## Purpose

This is a new product specification. 
To review go to **Files changed** and click **Display the rich Diff**

closes: https://github.com/datacite/tasks/issues/69

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (feature that would cause existing functionality to change)

- [ ] New Product (non-breaking change which adds a new service)

## Reviewer, please remember to:

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: product_specifications/DVjtom_67/readme.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+              And the repository is deployed to the web
+              And a third party visit the landing page with the tracking CODE correctly embeded.
+             Then the tracking CODE should trigger a call to the DataCite Usage Stats Processing Service
+              And the triggered call should include the all data needed for usage stats processing. 
+
+        Scenario: Installing the Usage stats tracker incorrectly
+            Given that the user has a Usage stats tracker CODE
+              And that the user has access to a source code of a repository with DOIs  
+              When the user includes the the Usage stats tracker CODE but either the tracking CODE or the DOI name are missing the repository source code
+             Then an Error Message in the developer tools should be displaying
```
Comment (sarala) [inline product_specifications/DVjtom_67/readme.md]: Is this done by us or plausible?

File: product_specifications/DVjtom_67/readme.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+This feature is implemented behind the `usage-processing-service` feature flag and disabled by default.
+Once we've confirmed the feature is deemed stable, we remove the feature flag to publish the feature as GA.
+
+## Technical Feasibility
+
+There are a few open questions about technical feasibility:
+
+- How can plausible be used display usage stats? 
+- How can relaiably process usage stats according to the CoP and produce similar stats as with the weblogs processing?
```
Comment (sarala) [inline product_specifications/DVjtom_67/readme.md]: I would like to see ongoing infrastructure cost estimates for this work.

File: product_specifications/DVjtom_67/readme.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+             When a user visits to the Usage Tracking Dashboard Tab
+             Then the user should be able to see the usage stats in different views.
+
+        Scenario: Displaying Usage Stats from Tracker in SUSHI Reports
+            Given that usage stats exist for a Repository on a given month
+             When a user visits to the enpoint '/reports/{report-id}'
+             Then the user should be able to download a usage report (SUSHI JSON) 
+
+```
+
```
Comment (sarala) [inline product_specifications/DVjtom_67/readme.md:158]: Should we mention how long we will keep the information and which data are we going to store long term?

File: product_specifications/DVjtom_67/readme.md (hunk 1 above)
Comment (kjgarza) [reply product_specifications/DVjtom_67/readme.md]: by "our" tracker. We have to modify not only the plausible hosted service but also the tracker.

File: product_specifications/DVjtom_67/readme.md (hunk 3 above)
Comment (kjgarza) [reply product_specifications/DVjtom_67/readme.md:158]: I was thinking about the same. According to COUNTER we need to maintain at least current year plus the previous two-years (https://www.projectcounter.org/frequently-asked-questions-faqs-counter-code-practice-release-5/). That being said, the bibliometricians might have a different perspective as they might want to keep the data for further studies.

File: product_specifications/DVjtom_67/readme.md (hunk 2 above)
Comment (kjgarza) [reply product_specifications/DVjtom_67/readme.md]: We can do some guess work but I think we will need a tech spec to be able to calculate that.

Comment (sarala) [reply product_specifications/DVjtom_67/readme.md]: Makes sense

**Review by sarala — APPROVED** [review]
All looks good.

**Comment by kjgarza** [comment]
@sarala I made further changes based on the product validation session, mostly on three areas:

1. Enable multiple and unintrusive ways to detect the DOI on the landing page. This is would ease the installation of the code without much development needed. 
2. A deactivation flow, from when the user decides to drop the service
3. URL-DOI check to discard counts from when the code is posted on the wrong landing page.

**Review by sarala — APPROVED** [review]
All looks good. In the feedback session, some were worried about not being able to execute javascript. I guess, we can only have a good idea after having a prototype. 

Do we foresee any security issues with using 3rd party tracking? I think this was also raised during the feedback session.

**Comment by kjgarza** [comment]
> some were worried about not being able to execute javascript.

I think this refers to the fact that machines/API's cannot execute the JS tracker, this is true but that is a limititation of the general approach In the grant. There were also comments about user disabling JS, and that's fine, we cannot force users to be tracked.

> Do we foresee any security issues with using 3rd party tracking?

Not really, we are the only third-party tracking here and the fact of including the tracking on the page is that the repo wants to be tracked. The source code is open as well and the users can see what we are doing if they so wish.

**Pull request merged**